### PR TITLE
[DO-NOT-MERGE] Try to update Hive to 2.3.2

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

Check if there is any test failed.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Nice try! Could you fix the remaining failure?

**Comment by wangyum** [comment]
Yes, I'm doing it

File: sql/core/pom.xml [hunk 1]
```diff
@@ -86,15 +86,17 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>org.apache.hive</groupId>
+      <artifactId>hive-storage-api</artifactId>
+    </dependency>
```
Comment (dongjoon-hyun) [inline sql/core/pom.xml]: BTW, you should not introduce `hive` into `sql/core`.
Apache ORC should be used as Apache ORC.

File: sql/core/pom.xml [hunk 2]
```diff
@@ -86,15 +86,17 @@
...
     <dependency>
       <groupId>org.apache.orc</groupId>
       <artifactId>orc-core</artifactId>
-      <classifier>${orc.classifier}</classifier>
```
Comment (dongjoon-hyun) [inline sql/core/pom.xml]: Please don't remove this.

File: sql/core/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcColumnarBatchReader.java [hunk 3]
```diff
@@ -21,6 +21,9 @@
 import java.util.stream.IntStream;
 
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.common.type.HiveDecimal;
+import org.apache.hadoop.hive.ql.exec.vector.*;
+import org.apache.hadoop.hive.serde2.io.HiveDecimalWritable;
```
Comment (dongjoon-hyun) [inline sql/core/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcColumnarBatchReader.java]: Please revert this.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala [hunk 4]
```diff
@@ -371,7 +371,7 @@ abstract class SQLViewSuite extends QueryTest with SQLTestUtils {
   test("correctly handle ALTER VIEW") {
     withTable("jt2") {
       withView("testView") {
-        sql("CREATE VIEW testView AS SELECT id FROM jt")
+        sql("CREATE VIEW testView AS SELECT 1 as c1 FROM jt")
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/SQLViewSuite.scala]: It is unrelated change and looks wrong.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala [hunk 5]
```diff
@@ -2513,9 +2513,9 @@ abstract class DDLSuite extends QueryTest with SQLTestUtils {
             }.getMessage
             assert(e.contains("Found duplicate column(s)"))
           } else {
-            sql("ALTER TABLE t1 ADD COLUMNS (C1 string)")
+            sql("ALTER TABLE t1 ADD COLUMNS (C2 string)")
             assert(spark.table("t1").schema ==
-              new StructType().add("c1", IntegerType).add("C1", StringType))
+              new StructType().add("c1", IntegerType).add("C2", StringType))
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala]: You are intentionally removing `case sensitive` test.

**Comment by dongjoon-hyun** [comment]
In general, this should keep all existing test coverage. Otherwise, it will not considered as a feasible solution.

**Comment by HyukjinKwon** [comment]
just out of curiosity, what does DNM stands for?

**Comment by dongjoon-hyun** [comment]
I guess `Do Not Merge`. Actually, this is a try to see the result. :)

**Comment by gatorsmile** [comment]
This is a very interesting work. Thanks for your effort!

When we implement such a feature, please remember we have to ensure it is configurable.

**Comment by jerryshao** [comment]
@gatorsmile @cloud-fan what is our plan for such thing? Currently we're using our own forked version, which blocks the support of Hadoop 3 ([SPARK-18673](https://issues.apache.org/jira/browse/SPARK-18673)). I'm wondering if we're going to upgrade Hive like what is done in this PR, or we still uses forked branch, and patch that branch.

What is your thought and plan on it?

**Comment by jerryshao** [comment]
@wangyum can you please create a relative JIRA, so that we discuss on the JIRA. I think it is better to discuss and get a conclusion first.

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
> [error] /home/jenkins/workspace/SparkPullRequestBuilder/sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/session/HiveSessionImpl.java:825:  error: cannot find symbol
[error]     String lScratchDir = hiveConf.getVar(ConfVars.LOCALSCRATCHDIR);
[error]  

But HiveSessionImpl.java#L825 is:
```
FileUtils.forceDelete(sessionLogDir);
```

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
Thanks everyone, we can move to [SPARK-23710](https://issues.apache.org/jira/browse/SPARK-23710) to discuss.

**Comment by dongjoon-hyun** [comment]
@wangyum . 
Please read again my previous [comment](https://github.com/apache/spark/pull/20659/files#r172587976).

Although I support your efforts, Spark can provide ORC data source without `-Phive`.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcDeserializer.scala [hunk 6]
```diff
@@ -17,9 +17,9 @@
 
 package org.apache.spark.sql.execution.datasources.orc
 
+import org.apache.hadoop.hive.serde2.io.{DateWritable, HiveDecimalWritable}
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcDeserializer.scala]: Please revert this.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala [hunk 7]
```diff
@@ -17,9 +17,9 @@
 
 package org.apache.spark.sql.execution.datasources.orc
 
-import org.apache.orc.storage.ql.io.sarg.{PredicateLeaf, SearchArgument, SearchArgumentFactory}
-import org.apache.orc.storage.ql.io.sarg.SearchArgument.Builder
-import org.apache.orc.storage.serde2.io.HiveDecimalWritable
+import org.apache.hadoop.hive.ql.io.sarg.{PredicateLeaf, SearchArgument, SearchArgumentFactory}
+import org.apache.hadoop.hive.ql.io.sarg.SearchArgument.Builder
+import org.apache.hadoop.hive.serde2.io.HiveDecimalWritable
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilters.scala]: Please revert this.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcSerializer.scala [hunk 8]
```diff
@@ -17,11 +17,11 @@
 
 package org.apache.spark.sql.execution.datasources.orc
 
+import org.apache.hadoop.hive.common.`type`.HiveDecimal
+import org.apache.hadoop.hive.serde2.io.{DateWritable, HiveDecimalWritable}
 import org.apache.hadoop.io._
 import org.apache.orc.TypeDescription
 import org.apache.orc.mapred.{OrcList, OrcMap, OrcStruct, OrcTimestamp}
-import org.apache.orc.storage.common.`type`.HiveDecimal
-import org.apache.orc.storage.serde2.io.{DateWritable, HiveDecimalWritable}
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcSerializer.scala]: Please revert this.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala [hunk 9]
```diff
@@ -22,7 +22,7 @@ import java.sql.{Date, Timestamp}
 
 import scala.collection.JavaConverters._
 
-import org.apache.orc.storage.ql.io.sarg.{PredicateLeaf, SearchArgument}
+import org.apache.hadoop.hive.ql.io.sarg.{PredicateLeaf, SearchArgument}
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/orc/OrcFilterSuite.scala]: Please revert this.

**Comment by dongjoon-hyun** [comment]
I added more specific comments about ORC parts, @wangyum .

**Comment by wangyum** [comment]
retest this please

**Comment by Bidek56** [comment]
What will take to merge this PR in? It has passed all the tests.

**Comment by srowen** [comment]
@wangyum you can close this experiment?
